WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/history/ScanHistoryViewModel.kt

```diff
@@ -1,17 +1,44 @@
 package org.wordpress.android.ui.jetpack.scan.history
 
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.launch
 import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.scan.threat.ThreatModel
+import org.wordpress.android.fluxc.store.ScanStore
+import org.wordpress.android.fluxc.store.ScanStore.FetchScanHistoryPayload
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.viewmodel.ScopedViewModel
 import javax.inject.Inject
+import javax.inject.Named
 
-class ScanHistoryViewModel @Inject constructor() : ViewModel() {
+class ScanHistoryViewModel @Inject constructor(
+    private val scanStore: ScanStore,
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher
+) : ScopedViewModel(mainDispatcher) {
     private var isStarted = false
 
+    private val _threats = MutableLiveData<List<ThreatModel>>()
+    val threats: LiveData<List<ThreatModel>> = _threats
+
     lateinit var site: SiteModel
 
     fun start(site: SiteModel) {
         if (isStarted) return
         isStarted = true
         this.site = site
+        fetchScanHistory()
+    }
+
+    private fun fetchScanHistory() {
+        launch {
+            val result = scanStore.fetchScanHistory(FetchScanHistoryPayload(site))
+            if (result.isError) {
+                // TODO malinjir handle error
+            } else {
+                _threats.value = scanStore.getScanHistoryForSite(site)
```

Comment:
It seems the usecase won't contain anything else than this single line of code - do you think it's worth wrapping into a use case?

Reply:
I feel like usecase are good in 3 scenarios
1. Converting event bus to a coroutine
2. Wrapping a code so it can be easily reused somewhere else
3. Wrapping a code which is complex and would deserve standalone test suite

I think this case doesn't fit any of the scenarios - tbh even the `fetchScanHistory` doesn't fit, since I doubt it'll ever get reused. Wdyt?

Reply:
We can keep this one here, I agree with your reasoning. But I still feel we should have a usecase for `fetchScanHistory` to be consistent with the rest of the code. Either case, use your best judgement. I have no strong opinion.

Reply:
> But I still feel we should have a usecase for fetchScanHistory to be consistent with the rest of the code.

👍 I'll introduce it in the upcoming PRs.